core/capabilities/targets/write_target.go

```diff
@@ -122,47 +116,81 @@ func (cap *EvmWrite) Execute(ctx context.Context, request capabilities.Capabilit
 
 	// TODO: validate encoded report is prefixed with workflowID and executionID that match the request meta
 
-	// construct forwarder payload
-	calldata, err := forwardABI.Pack("report", common.HexToAddress(reqConfig.Address), inputs.Report, inputs.Signatures)
+	txMeta := make(map[string]interface{}) // TODO: Consider just using the TxMeta struct here and pass it to submitTransaction
```

Comment:
What is the reason for not using the TxMeta struct (or a simplified version of that struct)?

Reply:
I need to update this PR accordingly.